Source/Core/Core/GeckoCodeConfig.cpp

```diff
@@ -22,8 +23,23 @@ std::vector<GeckoCode> DownloadCodes(std::string gametdb_id, bool* succeeded, bo
   // TODO: Fix https://bugs.dolphin-emu.org/issues/11772 so we don't need this workaround
   const std::string protocol = use_https ? "https://" : "http://";
 
-  // codes.rc24.xyz is a mirror of the now defunct geckocodes.org.
-  std::string endpoint{protocol + "codes.rc24.xyz/txt.php?txt=" + gametdb_id};
+  std::string endpoint;
+
+  std::string GeckoURL = Config::GetBase(Config::MAIN_GECKO_URL);
+  if (GeckoURL.find("https://") != std::string::npos)
+  {
+    GeckoURL.erase(0, 8);
+  }
+
+  if (GeckoURL != "")
```

Comment:
```suggestion
  if (!GeckoURL.empty())
```

(or flip the conditional and use `if (GeckoURL.empty())`)